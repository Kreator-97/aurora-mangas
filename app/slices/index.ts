export { default as uiSlice } from './uiSlice'
export { default as authSlice } from './authSlice'
export { default as shoppingCartSlice } from './shoppingCartSlice'
