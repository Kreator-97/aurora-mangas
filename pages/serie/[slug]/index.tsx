import { GetStaticPaths, GetStaticProps, NextPage  } from 'next'
import Image from 'next/image'

import { AppLayout } from '../../../layouts'
import { Serie } from '../../../interfaces'
import { CardGrid, CardManga } from '../../../components'
import { dbSeries } from '../../../database'

interface Props {
  serie: Serie
}

const SeriePage: NextPage<Props> = ({serie}) => {

  return (
    <AppLayout title={ serie.name } maxWidth='md'>
      <h1 className='text-center text-2xl py-4'>{ serie.name }</h1>
      <Image
        src={ serie.imgURL }
        width={360}
        height={180}
        layout="responsive"
        className='object-cover'
        alt={serie.name}
      />
      <h2 className='text-xl text-center mt-4'>Sinopsis</h2>
      <p className='text-lg m-4'>
        {
          serie.sinopsis
        }
      </p>
      <h2 className='text-xl text-center mt-4 mb-2'>Información</h2>
      <div className='px-4'>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Nombre</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.name}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Author</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.author.name}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Estado de la obra</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.finished ? 'Finalizado' : 'En progreso' }</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Volumenes publicados</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.volumes.length}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Periodicidad</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.periodicy}</p>
        </div>
        <div className='grid grid-cols-2 hover:bg-accent mb-4'>
          <p className='border border-strokeLight border-solid px-2'>Géneros</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.genre}</p>
        </div>
        <h2 className='text-center text-2xl mb-4'>Lista de volúmenes</h2>
        {
          <CardGrid gridCols='sm:grid-cols-1 md:grid-cols-2'>
            {
              serie.volumes.map((manga) => {
                return (
                  <CardManga manga={manga} key={manga.id}/>
                )
              })
            }
          </CardGrid>
        }
      </div>
    </AppLayout>
  )
}

export const getStaticPaths: GetStaticPaths = async (ctx) => {
  const slugs = await dbSeries.getAllSlugs()

  const paths = slugs.map(slug => {
    return { params: slug }
  })

  return {
    paths,
    fallback: 'blocking'
  }
}

export const getStaticProps: GetStaticProps = async (ctx) => {
  const slug = ctx.params?.slug?.toString()
  const serie = await dbSeries.getSerieBySlug(slug || '')

  return {
    props: {
      serie
    },
    revalidate: 86400
  }
}

export default SeriePage
