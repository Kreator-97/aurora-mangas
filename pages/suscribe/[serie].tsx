import { NextPage } from 'next'
import Image from 'next/image'
import { FaCcPaypal } from 'react-icons/fa'

import { Serie } from '../../interfaces'
import { AppLayout } from '../../layouts'
import { series } from '../index'

interface Props {
  serie: Serie;
}

const serie = series[0]

const SuscribeSeriePage: NextPage<Props> = () => {
  return (
    <AppLayout title={ `${serie.name} | subscripción` } maxWidth='lg'>
      <h1 className='text-xl text-center py-2'>{ serie.name }</h1>
      <Image
        src={serie.imgURL }
        width={360}
        height={180}
        layout="responsive"
        className='object-cover'
      />
      <h2 className='text-xl text-center mt-4'>Sipnosis</h2>
      <p className='text-lg m-4'>
        {
          serie.sipnosis
        }
      </p>
      <h2 className='text-xl text-center mt-4 mb-2'>Información</h2>
      <div className='px-4'>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Nombre</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.name}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Author</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.author.name} ({serie.author.birthDate})</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Volumenes</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.volumes.length}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Precio unitario</p>
          <p className='border border-strokeLight border-solid px-2'>$ {serie.unitPrice}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Periodicidad</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.periodicy}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Géneros</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.genre.join(', ')}</p>
        </div>
      </div>
      
      <section className='p-4 flex flex-col gap-y-2 items-center max-w-screen-sm mx-auto'>
        <h2 className='text-center text-2xl text-success'>Suscripción</h2>
        <p className='text-base text-success text-center'>
        Suscríbete y obtén todos los volúmenes en su fecha de publicación sin costo por envío + 5% de descuento
        </p>
        <button className='btn bg-success-gradient w-full text-lg'>Suscríbete por $3698</button>
        <p className='text-center'>Pago seguro procesado a través de:</p>
        <FaCcPaypal size={64}/>
      </section>
    </AppLayout>
  )
}

export default SuscribeSeriePage
