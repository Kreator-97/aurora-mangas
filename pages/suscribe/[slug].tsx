import { NextPage, GetServerSideProps } from 'next'
import Image from 'next/image'
import { FaCcPaypal } from 'react-icons/fa'

import { AppLayout } from '../../layouts'
import { dbSeries } from '../../database'
import { formatPrice } from '../../util'
import { Serie } from '../../interfaces'

interface Props {
  serie: Serie;
}

const SuscribeSeriePage: NextPage<Props> = ({serie}) => {
  return (
    <AppLayout title={ `${serie.name} | subscripción` } maxWidth='lg'>
      <h1 className='text-xl text-center py-2'>{ serie.name }</h1>
      <Image
        src={serie.imgURL }
        width={360}
        height={180}
        layout="responsive"
        className='object-cover'
      />
      <h2 className='text-xl text-center mt-4'>Sipnosis</h2>
      <p className='text-lg m-4'>
        {
          serie.sinopsis
        }
      </p>
      <h2 className='text-xl text-center mt-4 mb-2'>Información</h2>
      <div className='px-4'>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Nombre</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.name}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Author</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.author.name}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Volumenes</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.volumes.length}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Periodicidad</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.periodicy}</p>
        </div>
        <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
          <p className='border border-strokeLight border-solid px-2'>Géneros</p>
          <p className='border border-strokeLight border-solid px-2'>{serie.genre}</p>
        </div>
      </div>
      
      <section className='p-4 flex flex-col gap-y-2 items-center max-w-screen-sm mx-auto'>
        <h2 className='text-center text-2xl text-success'>Suscripción</h2>
        <p className='text-base text-success text-center'>
          Suscríbete y obtén todos los volúmenes en su fecha de publicación sin costo por envío
        </p>
        <button
          className='btn bg-success-gradient w-full text-lg'
        >
          { 
            serie.volumes[0]
              ? `Suscribete por ${formatPrice(serie.volumes[0].price)}`
              : 'Suscribete proximamente'
          }
        </button>
        <p>Cargo recurrente {serie.periodicy.toLowerCase()}</p>
        <p className='text-center'>Pago seguro procesado a través de:</p>
        <FaCcPaypal size={64}/>
      </section>
    </AppLayout>
  )
}

export default SuscribeSeriePage

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const slug = ctx.params?.slug?.toString()

  const serie = await dbSeries.getSerieBySlug(slug || '')

  if( !serie ) {
    return {
      notFound: true
    }
  }

  if( !serie.isNewRelease ) {
    return {
      redirect: {
        destination: '/',
        permanent: false
      }
    }
  }

  return {
    props: {
      serie
    }
  }
}
