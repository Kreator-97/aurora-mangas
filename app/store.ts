import { configureStore } from '@reduxjs/toolkit'
import { uiSlice, authSlice, shoppingCartSlice } from './slices'

export const store = configureStore({
  reducer: {
    ui: uiSlice,
    auth: authSlice,
    cart: shoppingCartSlice
  },
})

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch
