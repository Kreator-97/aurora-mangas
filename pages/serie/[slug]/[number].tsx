import { FC, useState } from 'react'
import { GetServerSideProps } from 'next'
import Image from 'next/image'
import Link from 'next/link'

import { addItem } from '../../../app/slices/shoppingCartSlice'
import { AppLayout } from '../../../layouts'
import { dbLocal } from '../../../util'
import { dbMangas } from '../../../database'
import { Manga } from '../../../interfaces'
import { openShoppingCart } from '../../../app/slices/uiSlice'
import { SelectAmount } from '../../../components'
import { useAppSelector, useAppDispatch } from '../../../app/hooks'


interface Props {
  manga: Manga;
}

const MangaPage:FC<Props> = ({manga}) => {
  const dispatch = useAppDispatch()
  const { cart } = useAppSelector(state => state)

  const [ counter, setCounter] = useState(1)
  
  const title = `${manga.serie.name} #${manga.number}`

  const onAddToCart = () => {
    dispatch( addItem({amount: counter, product: manga}))
    dispatch( openShoppingCart() )
    dbLocal.addItemInLocal(cart, {amount: counter, product: manga})
  }

  return (
    <AppLayout title={title} maxWidth='md'>
      <div className='grid grid-cols-1 sm:grid-cols-2 py-4'>
        <Image
          src={manga.imgURL}
          alt={title}
          layout="responsive"
          width={ 170 }
          height={ 255 }
          className="object-contain cursor-pointer animate__animated animate__backInLeft"
        />
        <div className='px-4 py-2'>
          <h1 className='text-3xl mb-4'><Link href={`/serie/${manga.serie.slug}`} >{title}</Link></h1>
          <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
            <p className='border border-strokeLight border-solid px-2'>Author</p>
            <p className='border border-strokeLight border-solid px-2'>{manga.serie.author.name}</p>
          </div>
          <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
            <p className='border border-strokeLight border-solid px-2'>Precio</p>
            <p className='border border-strokeLight border-solid px-2'>${manga.price}</p>
          </div>
          <div className='grid grid-cols-2 mb-2 hover:bg-accent'>
            <p className='border border-strokeLight border-solid px-2'>Publicado en</p>
            <p className='border border-strokeLight border-solid px-2'>{manga.published}</p>
          </div>

          <SelectAmount
            initial={1}
            onDecrement={(amount) => setCounter(amount)}
            onIncrement={(amount) => setCounter(amount)}
          />

          <button
            className='btn bg-accent w-full'
            onClick={ () => onAddToCart()}
          >
            Agregar al carrito
          </button>
        </div>
      </div>
    </AppLayout>
  )
}

export default MangaPage

export const getServerSideProps: GetServerSideProps = async (ctx) => {
  const serieName = ctx.query.slug?.toString()
  const number = ctx.query.number?.toString()

  const manga = await dbMangas.getMangaBySerieAndNumber(serieName || '', number || '')

  if( !manga ) {
    return {
      notFound: true
    }
  }

  return {
    props: {
      manga 
    }
  }
}
