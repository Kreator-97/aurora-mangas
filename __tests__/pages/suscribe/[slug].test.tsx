import { render, screen } from '@testing-library/react'
import { default as SuscribeSeriePage } from '../../../pages/suscribe/[slug]'
import { mangas, series } from '../../fixtures/db'

import { useSession } from 'next-auth/react'
import { Provider } from 'react-redux'
import { store } from '../../../app/store'

jest.mock('next-auth/react')
jest.mock('next/router', () => ({
  useRouter: () => ({query: { q: '' }})
}))

describe('tests on SuscribeSerie page', () => {
  (useSession as jest.Mock).mockReturnValue(() => ({
    status: 'unauthenticated',
    data: null,
  }))

  test('should to match with snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <SuscribeSeriePage serie={series[0]} />
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })

  test('should to show "Suscribete proximamente" button if there are not volumes published on series', () => {
    render(
      <Provider store={store}>
        <SuscribeSeriePage serie={series[0]} />
      </Provider>
    )

    expect( screen.getByText('Suscribete proximamente') ).toBeInTheDocument
  })

  test('should to show "Suscribete por" button if there are prices on volumes', () => {
    const manga = mangas[0]
    const serie = series[0]
    serie.volumes.push(manga)
    render(
      <Provider store={store}>
        <SuscribeSeriePage serie={serie} />
      </Provider>
    )

    expect( screen.getByText(/Suscribete por/) ).toBeInTheDocument
  })
})

