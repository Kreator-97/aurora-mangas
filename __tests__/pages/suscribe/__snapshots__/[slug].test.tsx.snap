// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests on SuscribeSerie page should to match with snapshot 1`] = `
<div>
  <nav
    class="container"
  >
    <div
      class="content"
    >
      <svg
        color="white"
        cursor="pointer"
        data-testid="menu"
        fill="currentColor"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        style="color: white;"
        viewBox="0 0 20 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
          fill-rule="evenodd"
        />
      </svg>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 150px; height: 30px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          alt="Aurora Mangas Logo"
          class="cursor-pointer"
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <div
        class="relative cursor-pointer"
        data-testid="cart-menu"
      >
        <svg
          color="white"
          cursor="pointer"
          fill="none"
          height="32"
          stroke="currentColor"
          stroke-width="0"
          style="color: white;"
          viewBox="0 0 24 24"
          width="32"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M3 3h2l.4 2M7 13h10l4-8H5.4M7 13L5.4 5M7 13l-2.293 2.293c-.63.63-.184 1.707.707 1.707H17m0 0a2 2 0 100 4 2 2 0 000-4zm-8 2a2 2 0 11-4 0 2 2 0 014 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
        <span
          class="w-6 h-6 text-center font-semibold bg-accentDark bg-opacity-95 rounded-full absolute -top-2 -right-2"
        >
          0
        </span>
      </div>
      <form
        class="flex col-span-2"
        data-testid="search-form"
      >
        <input
          class="w-full rounded border-cyan-400 border p-1 text-sm text-dark"
          placeholder="Encuentra tu manga favorito. Ejemplo: one punch man #1"
          type="search"
          value=""
        />
      </form>
      <svg
        color="white"
        cursor="pointer"
        fill="none"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        style="color: white;"
        viewBox="0 0 24 24"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M8 16l2.879-2.879m0 0a3 3 0 104.243-4.242 3 3 0 00-4.243 4.242zM21 12a9 9 0 11-18 0 9 9 0 0118 0z"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
        />
      </svg>
    </div>
  </nav>
  <div
    class="min-h-screen max-w-[360px] bg-dark fixed top-0 left-0 w-full z-50 p-4 transition-transform -translate-x-full"
  >
    <div
      class="flex justify-between items-center mb-2"
    >
      <div
        class="w-8"
      />
      <h2
        class="text-2xl"
      >
        Bienvenido
      </h2>
      <svg
        class="cursor-pointer"
        data-testid="close-icon"
        fill="currentColor"
        height="32"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
    <p
      class="text-center mb-2"
    >
      Ingresa para gestionar tu cuenta
    </p>
    <div
      class="flex gap-2 mb-4"
    >
      <button
        class="btn bg-accent w-full"
      >
        Ingresar
      </button>
      <button
        class="btn ghost w-full"
      >
        Crear cuenta
      </button>
    </div>
  </div>
  <div
    class="fixed top-0 right-0 min-h-screen h-screen min-w-[360px] bg-dark z-50 transition-transform translate-x-full"
  >
    <div
      class="flex justify-between items-center border-b border-strokeLight p-4"
      style="height: 65px;"
    >
      <svg
        cursor="pointer"
        data-testid="close-shopping-cart-icon"
        fill="currentColor"
        height="28"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 20 20"
        width="28"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M4.293 4.293a1 1 0 011.414 0L10 8.586l4.293-4.293a1 1 0 111.414 1.414L11.414 10l4.293 4.293a1 1 0 01-1.414 1.414L10 11.414l-4.293 4.293a1 1 0 01-1.414-1.414L8.586 10 4.293 5.707a1 1 0 010-1.414z"
          fill-rule="evenodd"
        />
      </svg>
      <h2
        class="text-center text-2xl"
      >
        Carrito de compra
      </h2>
      <div
        class="w-7"
      />
    </div>
    <div
      class="flex flex-col gap-4 items-center overflow-y-scroll cartlist-items"
      style="padding-top: 1rem; padding-bottom: 1rem;"
    >
      <h2
        class="text-center"
      >
        No hay elementos en el carrito de compra
      </h2>
    </div>
  </div>
  <main
    class="main"
    style="max-width: var(--lg);"
  >
    <h1
      class="text-xl text-center py-2"
    >
      Inuyasha
    </h1>
    <span
      style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
    >
      <span
        style="box-sizing: border-box; display: block; background: none; opacity: 1; border: 0px; margin: 0px; padding: 50% 0px 0px 0px;"
      />
      <img
        class="object-cover"
        data-nimg="responsive"
        decoding="async"
        src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
        style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
      />
      <noscript />
    </span>
    <h2
      class="text-xl text-center mt-4"
    >
      Sipnosis
    </h2>
    <p
      class="text-lg m-4"
    >
      InuYasha es el nombre de un demonio que en su día sellaron para que no provocara más problemas. Pero Kagome, fruto del destino, consigue atravesar la barrera mágica que separa el mundo actual con el de las guerras civiles, y quita el sello a Inu Yasha debido a la amenaza de cierto monstruo. Kagome posee la joya de las cuatro almas, que permite ganar gran poder, y ella será fruto del recelo de muchos monstruos en busca de ella. Pero la joya se ha roto en mil y un pedazos, e Inu Yasha y Kagome partirán en busca de todos los trozos de la joya.
    </p>
    <h2
      class="text-xl text-center mt-4 mb-2"
    >
      Información
    </h2>
    <div
      class="px-4"
    >
      <div
        class="grid grid-cols-2 mb-2 hover:bg-accent"
      >
        <p
          class="border border-strokeLight border-solid px-2"
        >
          Nombre
        </p>
        <p
          class="border border-strokeLight border-solid px-2"
        >
          Inuyasha
        </p>
      </div>
      <div
        class="grid grid-cols-2 mb-2 hover:bg-accent"
      >
        <p
          class="border border-strokeLight border-solid px-2"
        >
          Author
        </p>
        <p
          class="border border-strokeLight border-solid px-2"
        >
          Rumiko Takahashi
        </p>
      </div>
      <div
        class="grid grid-cols-2 mb-2 hover:bg-accent"
      >
        <p
          class="border border-strokeLight border-solid px-2"
        >
          Volumenes
        </p>
        <p
          class="border border-strokeLight border-solid px-2"
        >
          0
        </p>
      </div>
      <div
        class="grid grid-cols-2 mb-2 hover:bg-accent"
      >
        <p
          class="border border-strokeLight border-solid px-2"
        >
          Periodicidad
        </p>
        <p
          class="border border-strokeLight border-solid px-2"
        >
          MENSUAL
        </p>
      </div>
      <div
        class="grid grid-cols-2 mb-2 hover:bg-accent"
      >
        <p
          class="border border-strokeLight border-solid px-2"
        >
          Géneros
        </p>
        <p
          class="border border-strokeLight border-solid px-2"
        >
          ACTION, ADVENTURE, FANTASY, ROMANCE
        </p>
      </div>
    </div>
    <section
      class="p-4 flex flex-col gap-y-2 items-center max-w-screen-sm mx-auto"
    >
      <h2
        class="text-center text-2xl text-success"
      >
        Suscripción
      </h2>
      <p
        class="text-base text-success text-center"
      >
        Suscríbete y obtén todos los volúmenes en su fecha de publicación sin costo por envío
      </p>
      <button
        class="btn bg-success-gradient w-full text-lg"
      >
        Suscribete proximamente
      </button>
      <p>
        Cargo recurrente 
        mensual
      </p>
      <p
        class="text-center"
      >
        Pago seguro procesado a través de:
      </p>
      <svg
        fill="currentColor"
        height="64"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 576 512"
        width="64"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M186.3 258.2c0 12.2-9.7 21.5-22 21.5-9.2 0-16-5.2-16-15 0-12.2 9.5-22 21.7-22 9.3 0 16.3 5.7 16.3 15.5zM80.5 209.7h-4.7c-1.5 0-3 1-3.2 2.7l-4.3 26.7 8.2-.3c11 0 19.5-1.5 21.5-14.2 2.3-13.4-6.2-14.9-17.5-14.9zm284 0H360c-1.8 0-3 1-3.2 2.7l-4.2 26.7 8-.3c13 0 22-3 22-18-.1-10.6-9.6-11.1-18.1-11.1zM576 80v352c0 26.5-21.5 48-48 48H48c-26.5 0-48-21.5-48-48V80c0-26.5 21.5-48 48-48h480c26.5 0 48 21.5 48 48zM128.3 215.4c0-21-16.2-28-34.7-28h-40c-2.5 0-5 2-5.2 4.7L32 294.2c-.3 2 1.2 4 3.2 4h19c2.7 0 5.2-2.9 5.5-5.7l4.5-26.6c1-7.2 13.2-4.7 18-4.7 28.6 0 46.1-17 46.1-45.8zm84.2 8.8h-19c-3.8 0-4 5.5-4.2 8.2-5.8-8.5-14.2-10-23.7-10-24.5 0-43.2 21.5-43.2 45.2 0 19.5 12.2 32.2 31.7 32.2 9 0 20.2-4.9 26.5-11.9-.5 1.5-1 4.7-1 6.2 0 2.3 1 4 3.2 4H200c2.7 0 5-2.9 5.5-5.7l10.2-64.3c.3-1.9-1.2-3.9-3.2-3.9zm40.5 97.9l63.7-92.6c.5-.5.5-1 .5-1.7 0-1.7-1.5-3.5-3.2-3.5h-19.2c-1.7 0-3.5 1-4.5 2.5l-26.5 39-11-37.5c-.8-2.2-3-4-5.5-4h-18.7c-1.7 0-3.2 1.8-3.2 3.5 0 1.2 19.5 56.8 21.2 62.1-2.7 3.8-20.5 28.6-20.5 31.6 0 1.8 1.5 3.2 3.2 3.2h19.2c1.8-.1 3.5-1.1 4.5-2.6zm159.3-106.7c0-21-16.2-28-34.7-28h-39.7c-2.7 0-5.2 2-5.5 4.7l-16.2 102c-.2 2 1.3 4 3.2 4h20.5c2 0 3.5-1.5 4-3.2l4.5-29c1-7.2 13.2-4.7 18-4.7 28.4 0 45.9-17 45.9-45.8zm84.2 8.8h-19c-3.8 0-4 5.5-4.3 8.2-5.5-8.5-14-10-23.7-10-24.5 0-43.2 21.5-43.2 45.2 0 19.5 12.2 32.2 31.7 32.2 9.3 0 20.5-4.9 26.5-11.9-.3 1.5-1 4.7-1 6.2 0 2.3 1 4 3.2 4H484c2.7 0 5-2.9 5.5-5.7l10.2-64.3c.3-1.9-1.2-3.9-3.2-3.9zm47.5-33.3c0-2-1.5-3.5-3.2-3.5h-18.5c-1.5 0-3 1.2-3.2 2.7l-16.2 104-.3.5c0 1.8 1.5 3.5 3.5 3.5h16.5c2.5 0 5-2.9 5.2-5.7L544 191.2v-.3zm-90 51.8c-12.2 0-21.7 9.7-21.7 22 0 9.7 7 15 16.2 15 12 0 21.7-9.2 21.7-21.5.1-9.8-6.9-15.5-16.2-15.5z"
        />
      </svg>
    </section>
  </main>
  <footer
    class="min-h-[150px] border-t border-solid border-stroke"
  >
    <h2
      class="text-xl md:text-3xl text-center p-4 mt-8"
    >
      Síguenos en nuestras redes sociales
    </h2>
    <div
      class="flex justify-center gap-8"
    >
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 32px; height: 32px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
      <span
        style="box-sizing: border-box; display: inline-block; overflow: hidden; width: 36px; height: 36px; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: relative;"
      >
        <img
          data-nimg="fixed"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
        />
        <noscript />
      </span>
    </div>
  </footer>
</div>
`;
